### PR TITLE
update stable diffusion script and doc

### PR DESCRIPTION
**Pull request by tianleiwu**
### Description
Update script: 
(1) change some float16 verbose logging to debug level.
(2) Let requirements-cuda.txt includes requirements.txt
(3) Use an environment variable ORT_DISABLE_TRT_FLASH_ATTENTION=1 to avoid black image in 2.1 model. Update benchmark and doc.
(4) Update document to include command lines to build ORT rocm from source.
(5) Update optimize_pipeline.py so that user can disable packed qkv/kv from command line options.
(6) Update document to use torch < 2.0 for onnx export.

### CODE REVIEW

File: onnxruntime/python/tools/transformers/models/stable_diffusion/README.md [hunk 1]
```diff
@@ -74,90 +74,116 @@ Below is an example to optimize Stable Diffusion 1.5 in Linux. For Windows OS, p
...
+        python3-dev
+pip3 install numpy packaging "wheel>=0.35.1"
+wget --quiet https://github.com/Kitware/CMake/releases/download/v3.26.3/cmake-3.26.3-linux-x86_64.tar.gz
+tar zxf cmake-3.26.3-linux-x86_64.tar.gz
+export PATH=${PWD}/cmake-3.26.3-linux-x86_64/bin:${PATH}
+
+git clone https://github.com/microsoft/onnxruntime
+cd onnxruntime
+sh build.sh --config Release --cmake_extra_defines onnxruntime_USE_COMPOSABLE_KERNEL=ON  \
+            --use_rocm --rocm_home /opt/rocm --rocm_version 5.4.2 --build_wheel --allow_running_as_root
```
Comment (cloudhan) [inline onnxruntime/python/tools/transformers/models/stable_diffusion/README.md:137]: Lets make it a bit shorter:

sh build.sh --config Release --use_rocm --rocm_home /opt/rocm --rocm_version 5.4.2 --build_wheel

File: onnxruntime/python/tools/transformers/models/stable_diffusion/README.md [hunk 2]
```diff
@@ -74,90 +74,116 @@ Below is an example to optimize Stable Diffusion 1.5 in Linux. For Windows OS, p
...
-Please follow the [official docs](https://onnxruntime.ai/docs/build/eps.html#amd-rocm) to build ONNXRuntime from source.
+You can also follow the [official docs](https://onnxruntime.ai/docs/build/eps.html#amd-rocm) to build with docker.
 
 ### Export ONNX pipeline
+This step will export stable diffusion 1.5 to ONNX model in float32 using script from diffusers.
 
-This step will export stable diffusion 1.5 to ONNX model in float32 using script from diffusers. Before running the script, you need to be logged in via `huggingface-cli login`.
+It is recommended to use PyTorch 1.12.1 or 1.13.1 in this step. Using PyTorch 2.0 will encounter issue in exporting onnx.
+
+Before running the script, you need to be logged in via `huggingface-cli login`.
```
Comment (cloudhan) [inline onnxruntime/python/tools/transformers/models/stable_diffusion/README.md:148]: I don't think login is required anymore.

File: onnxruntime/python/tools/transformers/models/stable_diffusion/README.md [hunk 3]
```diff
@@ -74,90 +74,116 @@ Below is an example to optimize Stable Diffusion 1.5 in Linux. For Windows OS, p
...
 Example to benchmark the optimized pipeline with batch size 1 on CUDA EP:
 ```
-python -m onnxruntime.transformers.models.stable_diffusion.benchmark -p ./sd-v1-5-fp16/ -b 1
+python3 -m onnxruntime.transformers.models.stable_diffusion.benchmark -p ./sd -b 1
 ```
 
 On ROCm EP, use the following command instead:
 ```
-python -m onnxruntime.transformers.models.stable_diffusion.benchmark -p ./sd-v1-5-fp16/ -b 1 --tuning --provider=rocm
+python3 -m onnxruntime.transformers.models.stable_diffusion.benchmark -p ./sd -b 1 --tuning --provider rocm
```
Comment (cloudhan) [inline onnxruntime/python/tools/transformers/models/stable_diffusion/README.md:183]: Do we really need to emphasize `python3` and `pip3`, `python` and `pip` should work pretty well nowadays.

File: onnxruntime/python/tools/transformers/models/stable_diffusion/README.md (hunk 2 above)
Comment (tianleiwu) [reply onnxruntime/python/tools/transformers/models/stable_diffusion/README.md]: It is still required for benchmark to download torch model. So I moved it to benchmark section.

**Review by cloudhan — APPROVED** [review]
Otherwise, LGTM

**Pull request merged**